tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -100,27 +207,61 @@ def test_check_sfputil_eeprom(duthosts, enum_rand_one_per_hwsku_frontend_hostnam
-            if phy_intf in tested_physical_ports:
-                logging.info(
-                    "skip tested SFPs {} to avoid repeating operating physical interface {}".format(intf, phy_intf))
-                continue
-            tested_physical_ports.add(phy_intf)
-            logging.info("resetting {} physical interface {}".format(intf, phy_intf))
-            reset_result = duthost.command("{} {}".format(cmd_sfp_reset, intf))
-            assert reset_result["rc"] == 0, "'{} {}' failed".format(cmd_sfp_reset, intf)
-            time.sleep(5)
+    wait_after_restore_lpmode = 10
+    wait_after_ports_up = 60
+    wait_after_dom_config = 5
+    intfs_to_test, intfs = get_intfs_to_test(duthost,
+                                             conn_graph_facts,
+                                             enum_frontend_asic_index,
+                                             xcvr_skip_list)
+
+    # Form config interface cmds
+    cmds_down = []
+    cmds_up = []
+    cmds_disable_dom = []
+    cmds_enable_dom = []
+    for intf in intfs_to_test:
+        namespace_cmd_opt = get_namespace_cmd_option(
+            duthost, enum_frontend_asic_index)
+        cmds_down.append(cmd_config_intf_action.format(
+            namespace_cmd_opt, cmd_intf_shutdown, intf))
+        cmds_up.append(cmd_config_intf_action.format(
+            namespace_cmd_opt, cmd_intf_startup, intf))
+        cmds_disable_dom.append(cmd_config_intf_dom.format(
+            namespace_cmd_opt, intf, cmd_dom_disable))
+        cmds_enable_dom.append(cmd_config_intf_dom.format(
+            namespace_cmd_opt, intf, cmd_dom_enable))
+
+    # It's needed to shutdown ports before reset and startup ports after reset,
+    # to get config/state machine/etc replayed, so that the modules can be fully
+    # restored.
+    logging.info("Shutdown ports before sfp reset")
+    shutdown_result = duthost.shell_cmds(cmds=cmds_down)
+    assert any([r["rc"] for r in shutdown_result["results"]]) == 0, "Shutdown ports failed"
+    assert check_interface_status(duthost, intfs, expect_up=False)
+
+    logging.info("Check output of '{}' before sfp reset".format(cmd_sfp_show_lpmode))
+    lpmode_show = duthost.command(cmd_sfp_show_lpmode)
+    original_lpmode = parse_output(lpmode_show["stdout_lines"][2:])
+
+    logging.info("Disable DOM polling to avoid race condition during sfp reset")
+    disable_dom_result = duthost.shell_cmds(cmds=cmds_disable_dom)
+    assert any([r["rc"] for r in disable_dom_result["results"]]) == 0, "Disable DOM polling failed"
+    time.sleep(wait_after_dom_config)
+
+    for intf, phy_intf in intfs_to_test.items():
+        cmd_sfp_reset_intf = "{} {}".format(cmd_sfp_reset, intf)
+        logging.info("resetting {} physical interface {}".format(intf, phy_intf))
+        reset_result = duthost.command(cmd_sfp_reset_intf)
+        assert reset_result["rc"] == 0, "'{}' failed".format(cmd_sfp_reset_intf)
+        time.sleep(5)
     sleep_time = 60
```

Comment:
Is this this sleep still needed in addition to the above 5s sleep? Wondering since we are already polling for the link to be up as part of interface startup command.

Reply:
Removed the unnecessary sleep since we already have wait_util/polling logic